winch/codegen/src/codegen/mod.rs

```diff
@@ -31,6 +34,11 @@ where
 
     /// The MacroAssembler.
     pub masm: &'a mut M,
+
+    /// Stack frames for control flow.
+    // NB The 6 is set arbitrarily, we can adjust it as
+    // we see fit.
+    pub control_frames: SmallVec<[ControlStackFrame; 6]>,
```

Comment:
I don't want to bikeshed SmallVec sizes too much (we should just profile) but I'm curious, was this derived from some samples? I've seen block nesting go quite a bit deeper in real wasms; if `ControlStackFrame` isn't too large, and given that this is just one context struct for the length of the compilation, I don't see the harm in something like `64` or thereabouts?

Reply:
I didn't take any samples, so 6 was chosen arbitrarily, I'm not opposed to 64 though. I changed it to 64 and we can modify it later if needed.